### ISSUE TITLE
Add regression tests for the fix merged from PR 911

### ISSUE THREAD

**Issue by atamazov** [ROCm/MIOpen#930]
In order to provide the fix ASAP, #911 was merged in spite of missing regression tests, but we still need them. See also https://github.com/ROCmSoftwarePlatform/MIOpen/pull/911#issuecomment-837609892.

**Comment by junliume** [comment]
The issues found in #911 is not directly testable by MIOpenDriver, thus a holistic test from QA is needed. The information I get from @shaojiewang  and @carlushuang suggests that we should provide and ask QA to add the regression test to their procedure, instead of adding it to our CI regression pool.

**Comment by atamazov** [comment]
Can we can try to modify the driver?

**Comment by junliume** [comment]
> Can we can try to modify the driver?

+ @shaojiewang  @carlushuang to comment on whether MIOpenDriver can be modified to test this bugfix

This bug is found, fixed, and tested internally without QA involvement so far, I recommend submit a ticket for QA to add a holistic test in the meantime.

**Comment by carlushuang** [comment]
@junliume agree.
Hi @atamazov , this bug is reported [SWDEV-285232](https://ontrack-internal.amd.com/browse/SWDEV-285232), and this is reproduced only in E2E test with TF. I tried locally and can hardly reproduce by MIOpendriver, the reason I guess is there is not too much workload, and not too much memory acquire/release process, unlike TF. And even if the kernel read a out-of-bound memory address, it may not be a invalid value. Since this is a system problem, I guess the best approach is to let QA have a whole E2E test.

**Comment by atamazov** [comment]
We can specifically allocate a little more memory for the input tensor(s) than required, and fill that memory with NaNs. Perhaps this would make it possible to detect out-of-bound memory accesses more reliably, and sooner than QA, which is important. Opinions?

**Comment by carlushuang** [comment]
Yes this is a method for Nan caused by out-of-bound. I think the difficult thing is how to add this into either MIOpen or ctest, for this approach is a little bit hacky, perhaps need to add another flags/cmd_arg for this purpose. (or maybe default enable this while doing `make check` process)
I think this may contains 2 factors:

- for A/B matrix, allocate extra memory and fill that extra buffer with invalid float. 
- fill the C matrix with invalid float.

However, this may not make sure can detect all the Nan issues.

**Comment by atamazov** [comment]
@carlushuang 
> I think the difficult thing is how to add this into either MIOpen or ctest, for this approach is a little bit hacky, 

Yes, this is not easy to do.

> perhaps need to add another flags/cmd_arg for this purpose. (or maybe default enable this while doing `make check` process)

I would recommend the to make it the default (and I do not see any particular need to disable it).

>    * for A/B matrix, allocate extra memory and fill that extra buffer with invalid float.
>    * fill the C matrix with invalid float.

:+1: DEFINITELY YES.

> However, this may not make sure can detect all the Nan issues.

Of course. But shall we add this to your plans, what do you think?

**Comment by atamazov** [comment]
@shurale-nkn @Kirpich30000 Your opinion?

**Comment by carlushuang** [comment]
>> However, this may not make sure can detect all the Nan issues.

> Of course. But shall we add this to your plans, what do you think?

@atamazov I think until we see other patterns that can cause Nan issue, this *out-of-bound* plus *invalid float* is what we can do now to make sure not happen again.

**Comment by ppanchad-amd** [comment]
@atamazov Is this ticket still relevant?

**Comment by atamazov** [comment]
@ppanchad-amd Yes.

**Comment by JonathanLichtnerAMD** [comment]
Closing since MIOpen is being migrated to [ROCm/rocm-libraries](https://github.com/ROCm/rocm-libraries).  Please re-open there if you believe this is still an issue.
